Feat: 회원가입 페이지 API #35

kvkvd12 opened the pull request:
## PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [ ] 디자인 변경
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

## 반영 브랜치
feature/35-signupAPI -> API

## 변경 사항
1. 회원가입 
2. /users/signup/ 와 통신
3. 2차비밀번호 확인
4. 회원가입완료후 login.html로 이동
5. login.html에서 회원가입 버튼누르면 signup.html로 이동
6. gitignore에 venv추가

## 테스트 결과
변경 사항을 설명하는데 도움이 되는 스크린샷이나 참고 자료를 추가해주세요.
![image](https://user-images.githubusercontent.com/113074549/200186349-1c3507b9-8279-4845-a24a-cf086650e1a0.png)
![image](https://user-images.githubusercontent.com/113074549/200186373-435e0555-578b-4535-8248-b5eecb25db15.png)
![image](https://user-images.githubusercontent.com/113074549/200186401-4f2997c9-e1c0-4cf8-8835-acb601a61149.png)

LeeHyunji committed:
Merge pull request #23 from sparta-USD/feature/22-musicList

Design: 전체 노래 목록 UI 구현

kvkvd12 committed:
Merge branch 'API' of https://github.com/sparta-USD/Na-dle-Front

kvkvd12 committed:
Feat: 회원가입 페이지 API #35

1./users/signup/ 와 통신
2.2차비밀번호 확인
3.회원가입완료후 login.html로 이동
4.login.html에서 회원가입 버튼누르면 signup.html로 이동
5.gitignore에 venv추가

ssuin52 reviewed (approved):
회원가입 기능 연결하셨군요! 고생하셨습니다!!

LeeHyunji commented on static/js/signup.js:
```diff
@@ -0,0 +1,36 @@
+                "password2": password2
+            })
+        })
+        alert("회원가입 되었습니다.")
+        location.href = "http://127.0.0.1:5500/login.html";
+    }
```
이렇게 될경우 실제 request가 되지 않는 경우에도 회원가입되었다는 alert와 로그인페이지로 이동하게 됩니다.
response가 정상적으로 받아졌을 때 동작하도록 처리해야될것 같습니다.
마찬가지로 response에서 error가 발생했을대 error 메세지 처리도 필요할것 같습니다.

(참고)
```javascript
 const response = await fetch(....,{
...
})
.then(response => {
      if(!response.ok){
          throw new Error(`${response.status} 에러가 발생했습니다.`);    
      }
      return response.json()
  }).then(result => {
      alert("프로필이 수정되었습니다.")
      document.getElementById("profile_img").setAttribute("src","http://127.0.0.1:8000"+result['profile_image'])
  }).catch(error => {
      alert("프로필 수정에 실패하였습니다.\n자세한 내용은 관리자에게 문의해주세요.")
      console.warn(error.message)
  });
```

kvkvd12 replied:
에러처리 좀 더 완벽하게 해서 올리겠습니다.

kvkvd12 committed:
Fix: 회원가입 response 예외처리 #35

1. 빈칸이 존재할때 ( fullname, username, password, password2 ) 예외처리
2. 아이디가 이미 존재할때 예외처리

Close #35

kvkvd12 committed:
Fix: 회원가입 response 예외처리 #35

1. 아이디가 이미 존재할때 예외처리 ( response.status 띄워줌 )

Close #35
